client-ui/src/app/components/time-frame-picker/date-picker/date-picker.component.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, EventEmitter, Input, OnChanges, Output} from '@angular/core';
+import {NgbCalendar, NgbDateStruct} from "@ng-bootstrap/ng-bootstrap";
+import * as _ from "lodash";
+
+const MODEL_UPDATE_DELAY_ON_TYPING = 500 //millis
+
+@Component({
+  selector: 'date-picker',
+  templateUrl: './date-picker.component.html',
+  styleUrls: ['./date-picker.component.less']
+})
+export class DatePickerComponent implements OnChanges {
+
+  @Input() model: NgbDateStruct
+  @Input() minDate: NgbDateStruct
+  @Input() maxDate: NgbDateStruct
+
+  @Output() modelChange = new EventEmitter<NgbDateStruct>()
+
+  constructor(private calendar: NgbCalendar) {
+  }
+
+  valid: boolean
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'valid' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)